### ISSUE TITLE
DAP hangs/fails to connect if previous debug session was ended by quitting FS-UAE

### ISSUE THREAD

**Issue by dansalvato** [grahambates/uae-dap#1]
I'm on macOS using Neovim. To explain the issue, let me list a working and non-working scenario.

Working scenario:

1. Start a debug session. Node and FS-UAE start up.
2. End the session from the debugger by sending `terminate()`. FS-UAE and Node both quit.
3. Start a new debug session. Node and FS-UAE start up and work as expected.

Another working scenario:
1. Start a debug session. Node and FS-UAE start up.
2. Force quit Node, and then quit FS-UAE normally.
3. Start a new debug session. Node and FS-UAE start up and work as expected.

Non-working scenario:
1. Start a debug session. Node and FS-UAE start up.
4. End the session by quitting FS-UAE. FS-UAE and Node both quit.
5. (Optional) Restart Neovim to ensure nothing stale is going on with the debugger.
6. Start a new debug session. FS-UAE and Node start up, but FS-UAE hangs. A force quit is required on both FS-UAE and Node.
7. After force quitting Node and FS-UAE, start another debug session. Node and FS-UAE start up and work as expected.

Basically, if the debug session is ended by quitting FS-UAE, then the next session is guaranteed to fail. It's very strange because I confirmed that the Node process also ends when quitting FS-UAE. Is there a difference between how the DAP handles FS-UAE quitting versus `terminate()` from the debugger?

**Comment by grahambates** [comment]
I'm pretty sure we had this exact bug on Bartman's vscode-amiga-debug extension. I'll sort it.

**Comment by dansalvato** [comment]
I haven't had any hangs in quite a while at this point, so I'll go ahead and close this.
